src/tools/kata-ctl/src/arch/x86_64/mod.rs

```diff
@@ -61,4 +69,9 @@ mod arch_specific {
 
         Ok(())
     }
+
+    fn check_kernel_modules(_args: &str) -> Result<()> {
```

Comment:
@dborquez - When this function is implemented, it would call a generic function that will check if a kernel module is loaded (or if the feature is built into the kernel). Something like:

```rust
check_kernel_module_loaded("kvm")?;
check_kernel_module_loaded("kvm_intel")?;
```

That way, the PR for #6132 could also call this generic function. For example:

```rust
check_kernel_module_loaded("vhost_net")?;
```

In reality, this generic function will need to accept more than just the module name because we need to also check that module parameters are correct too (for example `kvm_intel` should be loaded with the `unrestricted_guest=Y` parameter for that module).

Reply:
@jodh-intel thank you, I have added a `KernelModule` type structure that wraps the _name_ of the module and a map with the _(parameter,value)_ pair. So `check_kernel_module_loaded()` accepts this type of data.